src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: Array<T>, key: keyof T): GroupsMap<T> {
+  const resOfArr: GroupsMap<T> = {};
```

Comment:
`resOfArr`? Change this variable name, cuz it is not entirely clear what is contained in the middle.
use `groupedItems` for example.